pkg/keystone/keystone.go

```diff
@@ -80,36 +81,37 @@ func (d *keystone) init() {
 	if viper.Get("keystone.username") != nil {
 		// force service logon to check validity early
 		// this will set d.providerClient
-		_, err := d.serviceKeystoneClient()
+		ctx := context.Background()
```

Comment:
the init function should probably passthru a ctx from higher up. If you are using cobra than that already provides one similar to req.Context(), otherwise you can generate one like https://github.com/sapcc/limes/blob/f8fc560b82c6826cad187cc7fb289dcc47faa7d5/main.go#L83

Reply:
viper doesn't have a context that i can see, so I implemented it more like the example. might not make as much sense, but attempted it.